components/ArchivePosts.php

```diff
@@ -0,0 +1,186 @@
+            ],
+            'currentPage' => [
+                'title'       => 'rainlab.blog::lang.settings.posts_pagination',
+                'description' => 'rainlab.blog::lang.settings.posts_pagination_description',
+                'type'        => 'string',
+                'default'     => '{{ :page }}',
+            ],
+            'postsPerPage' => [
+                'title'             => 'rainlab.blog::lang.settings.posts_per_page',
+                'type'              => 'string',
+                'validationPattern' => '^[0-9]+$',
+                'validationMessage' => 'rainlab.blog::lang.settings.posts_per_page_validation',
+                'default'           => '10',
+            ],
+            'noPostsMessage' => [
+                'title'        => 'rainlab.blog::lang.settings.posts_no_posts',
+                'description'  => 'rainlab.blog::lang.settings.posts_no_posts_description',
+                'type'         => 'string',
+                'default'      => 'No posts found',
+                'showExternalParam' => false
+            ],
+            'sortOrder' => [
+                'title'       => 'rainlab.blog::lang.settings.posts_order',
+                'description' => 'rainlab.blog::lang.settings.posts_order_description',
+                'type'        => 'dropdown',
+                'default'     => 'published_at desc'
+            ],
+            'categoryPage' => [
+                'title'       => 'rainlab.blog::lang.settings.posts_category',
+                'description' => 'rainlab.blog::lang.settings.posts_category_description',
+                'type'        => 'dropdown',
+                'default'     => 'blog/category',
+                'group'       => 'Links',
+            ],
+            'postPage' => [
+                'title'       => 'rainlab.blog::lang.settings.posts_post',
+                'description' => 'rainlab.blog::lang.settings.posts_post_description',
+                'type'        => 'dropdown',
+                'default'     => 'blog/post',
+                'group'       => 'Links',
+            ],
+        ];
+    }
+
+    public function getCategoryPageOptions()
+    {
+        return Page::sortBy('baseFileName')->lists('baseFileName', 'baseFileName');
+    }
+
+    public function getPostPageOptions()
+    {
+        return Page::sortBy('baseFileName')->lists('baseFileName', 'baseFileName');
+    }
+
+    public function getSortOrderOptions()
+    {
+        return BlogPost::$allowedSortingOptions;
+    }
+
+    public function onRun()
+    {
+        $this->prepareVars();
+        $this->posts = $this->page['posts'] = $this->listPosts($this->property('month'));
+
+        /*
+         * If the page number is not valid, redirect
+         */
+        if ($currentPageParam = $this->paramName('currentPage')) {
+            $currentPage = $this->property('currentPage');
+
+            if ($currentPage > ($lastPage = $this->posts->lastPage()) && $currentPage > 1)
```

Comment:
Please replace with the `{}` wrapper, we try to avoid the one line conditionals in October